### PR TITLE
Channel Sections

### PR DESCRIPTION
**Pull request by infinite-persistence**
- Test: `kp`
- See `overview.md` and commit messages for details.

### CODE REVIEW

**Comment by infinite-persistence** [comment]
@saltrafael, it's in `draft`, but reviewable. 
The channel finder is hardcoded to fake lighthouse results for now, but direct URLs (https://, lbry://) would still work to find a specific entry.

**Comment by infinite-persistence** [comment]
[28c7be2 ](https://github.com/OdyseeTeam/odysee-frontend/commit/28c7be20e0762996828ba3f6f9e05629f5730411)to [aa06482 ](https://github.com/OdyseeTeam/odysee-frontend/commit/aa06482b59ae3400ebb9da6a61e5033cd814b1df):
- The channel finder can now be tested.
- Updated based initial style-pass feedback.
- Temporarily removed the "Customized Sections" page since we only have 1 section type for now.  Featured Channels will now be edited directly from the Channels Tab.

**Comment by infinite-persistence** [comment]
[d9a4427 ](https://github.com/OdyseeTeam/odysee-frontend/commit/d9a4427af1b45c6d89b7f7f107a671c868fbc1cb)
- Use placeholder/skeleton in channel-finder while fetching.

**Comment by tzarebczan** [comment]
Screenshot_20220803-204403
Mobile menu could use some work - not sure if anything was done for this on creator memberships since that's adding a tag. It can be some overflow menu? 

Does YouTube show the tab when there's no featured channels? 

Didn't get a chance to test much, hope to do more tomorrow.

**Comment by infinite-persistence** [comment]
> _Mobile menu could use some work - not sure if anything was done for this on creator memberships since that's adding a tag. It can be some overflow menu?_

Yup, still remember your comment in notion. Forgot to ask Haffa about this. @saltrafael, was there work on this that I can cherry-pick?  Else, I'm just implement the overflow button.

> _Does YouTube show the tab when there's no featured channels?_

Yes, plus after a review with Rave, we've removed the "customize section" button in the channel page banner since there's only 1 section type for now, so the tab has to be there to allow owners to edit (although bringing back the "customized section" button is also a solution).

**Comment by tzarebczan** [comment]
I'm ok with always showing the section, was just thinking it could make mobile UX better in meantime until more set it up. We could show it just for channel owners, or if they exist..but no biggie. 

Should we limit to a single section until we have the ability to allow customizing (i.e move sections up and down)? 

- [ ] indicator to type more than 2 characters to get search results? Maybe use claim search before that? minor... 
- [ ] how hard would it be to support reordering here (could be drag and drop too, but needs to be intuitive)? I know you can reorder later in the edit page, but I wouldn't want to use that same publish page for this flow if possible. Maybe a dumbed-down / custom version of it if required, i.e don't show playlist icon / private. Looks like others can see this page too if they click the 3 button. 
image

- [ ] Not sure I like "unpublished" - I know we use that in other areas, but it may not be as intuitive here. I was thinking "unsaved" , but the selection screen says save.  "Not public"? 
- [ ] Not intuitive that you need to publish after - we probably don't need the whole playlist flow -but instead just show a "Make public" button which automatically calls collection create/update for all. I don't see what data I'd add here, title is really all you need? 
- [ ] update this wording with whatever we decide on vs "publish"
image

- [ ] probably don't want to show these in playlists section either 
- [ ] disable copy link option on channels page at least

**Comment by drewhancock** [comment]
image


- [ ] Instead of the channels shrinking after you add 8, they should stay the default size.

- [ ] Instead of the scroll bar, I think we should copy what YouTube does with the arrows. After three clicks, you can have the three button option to view the full list. Can look at this channel for reference: https://www.youtube.com/c/AngloLibertarian

- [ ] If you only have one section, it shouldn't be blocked off, it should take up the entire space. Like this: 

image

- [ ] If possible, we should get rid of this edit menu entirely. 

image

All the editing should be done here:

image

- [ ] We should have a way to reorder sections here 

image

**Comment by IronMaltese** [comment]
> Yes, plus after a review with Rave, we've removed the "customize section" button in the channel page banner since there's only 1 section type for now, so the tab has to be there to allow owners to edit (although bringing back the "customized section" button is also a solution).

Even though YouTube always shows this feature, odysee shouldn't show this tab unless the user explicitly adds channels to it, since this gives content creators more control over how minimal they want their channel to look in terms of features included on their page


Also for the end user, pressing a button that leads to nothing wastes a bit of time unnecessarily and regardless of how small this time waste is, it's avoidable and should be only available after channels are added

File: ui/component/channelFinder/entry/index.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import Entry from './view';
```
Comment (saltrafael) [inline ui/component/channelFinder/entry/index.js:1]: empty index.js needed?

File: ui/component/channelSections/SectionList/view.jsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    }
+  }
+
+  React.useEffect(() => {
+    if (!claimSearchResult) {
+      setIsClaimSearching(true);
+      doClaimSearch(options, { useAutoPagination: true }).finally(() => {
+        setTimeout(() => {
+          setIsClaimSearching(false);
+        }, 500); // Est. time for reducers to complete.
```
Comment (saltrafael) [inline ui/component/channelSections/SectionList/view.jsx:87]: Wouldn't be able to pass as callback function to run on reducers complete?

File: ui/component/claimPreviewTile/view.jsx [hunk 3]
```diff
@@ -95,7 +95,7 @@ function ClaimPreviewTile(props: Props) {
     swipeLayout = false,
     onHidden,
     pulse,
-    collectionFirstUrl,
+    // collectionFirstUrl, -- still useful?
```
Comment (saltrafael) [inline ui/component/claimPreviewTile/view.jsx:98]: Probably not, looks like playlist previews already redirect to the first item

File: static/app-strings.json [hunk 4]
```diff
@@ -2036,6 +2036,7 @@
   "Item added to Favorites": "Item added to Favorites",
   "Item removed from Favorites": "Item removed from Favorites",
   "Your publish is being confirmed and will be live soon": "Your publish is being confirmed and will be live soon",
+  "Your changes is being confirmed and will be live soon": "Your changes is being confirmed and will be live soon",
```
Comment (saltrafael) [inline static/app-strings.json:2039]: Typo

File: ui/scss/component/_channel.scss [hunk 5]
```diff
@@ -108,6 +108,12 @@ $actions-z-index: 2;
     font-weight: var(--font-weight-bold);
   }
 
+  .channel__edit-grp {
```
Comment (saltrafael) [inline ui/scss/component/_channel.scss:111]: Unused?

**Pull request merged**